src/module_wrap.cc

```diff
@@ -481,19 +480,72 @@ Maybe<uv_file> CheckFile(const URL& search,
   uv_fs_req_cleanup(&fs_req);
 
   if (is_directory) {
-    uv_fs_close(nullptr, &fs_req, fd, nullptr);
+    CHECK_EQ(0, uv_fs_close(nullptr, &fs_req, fd, nullptr));
     uv_fs_req_cleanup(&fs_req);
     return Nothing<uv_file>();
   }
 
   if (opt == CLOSE_AFTER_CHECK) {
-    uv_fs_close(nullptr, &fs_req, fd, nullptr);
+    CHECK_EQ(0, uv_fs_close(nullptr, &fs_req, fd, nullptr));
     uv_fs_req_cleanup(&fs_req);
   }
 
   return Just(fd);
 }
 
+const Environment::PackageConfig& GetPackageConfig(Environment* env,
+                                                   const std::string path) {
+  static const Environment::PackageConfig kEmptyPackage = { false, false, "" };
+
+  auto existing = env->package_json_cache.find(path);
+  if (existing != env->package_json_cache.end())
+    return existing->second;
+  Maybe<uv_file> check = CheckFile(path, LEAVE_OPEN_AFTER_CHECK);
+  if (check.IsNothing())
+    return kEmptyPackage;
+
+  Isolate* isolate = env->isolate();
+
+  v8::HandleScope handle_scope(isolate);
+
+  std::string pkg_src = ReadFile(check.FromJust());
+  uv_fs_t fs_req;
+  CHECK_EQ(0, uv_fs_close(nullptr, &fs_req, check.FromJust(), nullptr));
+  uv_fs_req_cleanup(&fs_req);
+
+  // It's not okay for the called of this method to not be able to tell
+  // whether an exception is pending or not.
+  TryCatch try_catch(isolate);
```

Comment:
The TryCatch means exceptions from the code below won't propagate.  Either remove this (I think you should) or drop the 'exception pending' comments below.

Reply:
So what we'd actually want is to throw a JS error for JSON parsing failure. But if the above is definitely not going to help with that sure, it can be removed.